python/essex-config/README.md

```diff
@@ -0,0 +1,106 @@
+# essex-configuration
+
+essex-configuration is a Python library for creating configuration objects that read configuration from different sources. This library supports reading from files, environment variables, and Azure Key Vault.
+
+## Installation
+
+To install the essex-configuration library, use pip:
+
+`pip install essex-configuration`
+
+## Usage
+
+Here’s an example of how to use the essex-configuration library to create a configuration object for connecting to a customer database:
+
+```python
+from typing import Annotated, ClassVar
+from essex_config import Config, ConfigurationField, FieldType
+from essex_config.sources import Source, EnvSource, FileSource, KeyvaultSource
+
+
+class CustomerDatabase(Config):
```

Comment:
Can configuration objects represent an object hierarchy? That would get us to parity with javascript tooling. e.g. a top-level app-config with sub-objects for database config, llm config, etc..

Reply:
Here you could do something like:

```python
class TopLevelConfig(BaseModel):

    @property  # or cached_property if you like
    def llm(self) -> LLMConfig:
        return LLMConfig.get_config()
    @property
    def other_config(self):
       ...
```

and then you can use TopLevelConfig as a singleton and access it with TopLevelConfig.get_instance().llm

But I don't see why if anywhere on the code you could do `LLMConfig.get_config()`.

Reply:
Let's chat on Thursday after GR day. I'd like to walk through some use cases and expand on my comments here. I think is a great start, I just want to make sure we have a flexible consumer abstraction